### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by adarachel**
# In this task, we worked with the real live data from the SpaceX API. We built a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.

## We implemented the following project requirements:

Config & basic setup
Create React App.
Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
Create routes and view components: rockets, missions, my profile. Add dragons [only if your team has 3 members]).
Use <NavLink /> for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
Create directories for all Redux state slice files:rockets, missions, dragons [only if your team has 3 members].
Redux: Fetch data and update Redux store
Upon first render fetch data from the SpaceX API endpoints:
Rockets: https://api.spacexdata.com/v4/rockets
Missions: https://api.spacexdata.com/v3/missions
Dragons: https://api.spacexdata.com/v4/dragons [only if your team has 3 members]
Once the data are fetched, dispatch an action to store the selected data in Redux store:

Technical set up**
Set up the repository on GitHub and use Gitflow.
Set up React and Redux.
Set up testing libraries(React Testing Library and Jest).
Do not preserve user data.

### CODE REVIEW

File: src/components/RocketsProfile.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const rockets = useSelector((state) => state.rockets);
+  const reservedRockets = rockets.filter((rocket) => rocket.reserved);
+  return (
+    <div className="reserve-container">
+      <h1 className="profile-rockets-title">My Rockets</h1>
+      <div className="profile-rockets">
+        {reservedRockets.length === 0 && <p className="no-reserve">No Rockets Reserved</p>}
+        {reservedRockets.map((rocket) => (
+          <div key={rocket.id} className="reserve-item">{rocket.rocket_name}</div>
+        ))}
```
Comment (VICTOR-CHUKWU) [inline src/components/RocketsProfile.js:14]: - [ ] You know you can add some features to this profile i.e. a user should be able to cancel a reservation from the profile. you can try your hands on it :+1:

File: src/redux/missions/missions.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  initialState,
+  reducers: {
+    toogleReserve: (state, { payload }) => {
+      state.missions = state.missions.map((mission) => { // eslint-disable-line no-param-reassign
+        if (mission.mission_id === payload) {
+          return { ...mission, reserved: !mission.reserved };
+        } return mission;
+      });
+    },
+  }, /* eslint-disable no-param-reassign */
```
Comment (VICTOR-CHUKWU) [inline src/redux/missions/missions.js:24]: - [ ] linters are here to help you write clean codes, try not to disable many while working on a project :clap:

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @adarachel  and @LoviMadaan , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] good use of Router

## what to consider
- [ ] try and make your app responsive

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/missions/missions.js (hunk 2 above)
Comment (adarachel) [reply src/redux/missions/missions.js:24]: Thanks!

**Pull request merged**